File: leetCode/twoSum/twoSum.js
const nums = [2,7,11,15], target = 9;

const twoSum = (nums, target) => {
  const hash = {};

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];

    if (diff in hash) return [hash[diff], i];

    hash[nums[i]] = i;
  }
}

console.log(twoSum(nums, target));

File: leetCode/twoSum/twoSum.py
nums, target = [2,7,11,15], 9

def twoSum(nums, target):
  hash = {}

  for i in range(len(nums)):
    diff = target - nums[i]

    if diff in hash:
      return [hash[diff], i]
    else:
      hash[nums[i]] = i

print(twoSum(nums,target))